test/v1/controllers/requestParsers/CreateAndAmendEmploymentExpensesRequestParserSpec.scala

```diff
@@ -0,0 +1,103 @@
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import play.api.libs.json.Json
+import support.UnitSpec
+import v1.mocks.validators.MockCreateAndAmendEmploymentExpensesValidator
+import v1.models.domain.Nino
+import v1.models.errors.{BadRequestError, ErrorWrapper, NinoFormatError, TaxYearFormatError}
+import v1.models.request.TaxYear
+import v1.models.request.createAndAmendEmploymentExpenses.{
+  CreateAndAmendEmploymentExpensesBody,
+  CreateAndAmendEmploymentExpensesRawData,
+  CreateAndAmendEmploymentExpensesRequest,
+  Expenses
+}
+
+class CreateAndAmendEmploymentExpensesRequestParserSpec extends UnitSpec {
+
+  val nino    = "AA123456B"
+  val taxYear = "2017-18"
+
+  implicit val correlationId: String = "a1e8057e-fbbc-47a8-a8b4-78d9f015c253"
+
+  private val requestBodyJson = Json.parse("""
+                                             |{
+                                             |    "expenses": {
+                                             |        "businessTravelCosts": 123.12,
+                                             |        "jobExpenses": 123.12,
+                                             |        "flatRateJobExpenses": 123.12,
+                                             |        "professionalSubscriptions": 123.12,
+                                             |        "hotelAndMealExpenses": 123.12,
+                                             |        "otherAndCapitalAllowances": 123.12,
+                                             |        "vehicleExpenses": 123.12,
+                                             |        "mileageAllowanceRelief": 123.12
+                                             |    }
+                                             |}""".stripMargin)
+
+  private val rawData: CreateAndAmendEmploymentExpensesRawData =
+    CreateAndAmendEmploymentExpensesRawData(nino = nino, taxYear = taxYear, body = requestBodyJson)
+
+  private val requestData = CreateAndAmendEmploymentExpensesRequest(
+    nino = Nino(nino),
+    taxYear = TaxYear.fromMtd(taxYear),
+    body = CreateAndAmendEmploymentExpensesBody(
+      Expenses(Some(123.12), Some(123.12), Some(123.12), Some(123.12), Some(123.12), Some(123.12), Some(123.12), Some(123.12))
+    )
+  )
+
+  trait Test extends MockCreateAndAmendEmploymentExpensesValidator {
+
+    lazy val parser = new CreateAndAmendEmploymentExpensesRequestParser(mockValidator)
+  }
+
+  "parse" should {
+
+    "return a request object" when {
+
+      "valid request data is supplied" in new Test {
+
+        MockCreateAndAmendEmploymentExpensesValidator.validate(rawData).returns(Nil)
+
+        parser.parseRequest(rawData) shouldBe Right(requestData)
+      }
+    }
+
+    "return an ErrorWrapper" when {
+
+      "a single validation error occurs" in new Test {
+        MockCreateAndAmendEmploymentExpensesValidator
+          .validate(rawData)
+          .returns(List(NinoFormatError))
+
+        parser.parseRequest(rawData) shouldBe
```

Comment:
can also assign all of these to `val result`s, just adds a bit of clarity/intended purpose to the tests